### PR TITLE
Create Transaction part in JavaScript #17

### LINKED ISSUE #17: Transaction service implementation
## Description
Based on the requirements outlined in PR #3, we need to implement a solution for Transaction service

## Acceptance criteria

1. All the functions from specification related to Transaction are implemented
2. Errors are handled too

Note: target main branch for your PRs

### PR DESCRIPTION
**Pull request by JCRigo86**
### Description

Completes the code with Transactions and UTXO implementation

### Related Issue

Fixes #17 

### Changes Made

Added the transaction.js and UTXO.js modules

### How to Test

JEST files to be finalised 

### By submitting this pull request, I confirm the following (please tick checkboxes):

- [x] Any new functions, methods, or classes are properly documented.
- [x] The commit messages are clear and descriptive.
- [x] The pull request title is concise and descriptive.
- [x] The pull request is targeted to the correct branch.

### CODE REVIEW

**Review by WaltervonGabain — APPROVED** [review]

File: src/transaction.js [hunk 1]
```diff
@@ -1 +1,138 @@
...
+        let balance = 0;
+        for (const utxo of this.utxos) {
+            balance += utxo.message;
+        }
+        const amount_enough = balance === spent;
+        return signature_valid && amount_enough;
+    }
+}
+
+class UnsignedTransaction {
```
Comment (rkokhatskyi) [inline src/transaction.js:68]: Can we split this into a different file as well as Coinbase class, better to keep the files shorter and organized

File: src/transaction.js [hunk 2]
```diff
@@ -1 +1,138 @@
...
+
+    sign(priv_key, password) {
+        return _sign(priv_key, password, this.getHash());
+    }
+}
+
+class Coinbase {
+    constructor(receiver) {
+        this.receiver_public_keys = [receiver];
+        this.messages = [50];
```
Comment (rkokhatskyi) [inline src/transaction.js:113]: What is value 50 about?

File: src/transaction.js [hunk 3]
```diff
@@ -1 +1,138 @@
...
+    isValid() {
+        return true;
+    }
+
+    getDict() {
+        const data = {
+            "receiver_public_keys": this.receiver_public_keys,
+            "messages": this.messages
+        };
+        return data;
```
Comment (rkokhatskyi) [inline src/transaction.js:132]: you can return object directly like you did in other class

File: src/transaction.js [hunk 4]
```diff
@@ -1 +1,138 @@
...
+    }
+
+    getDict() {
+        const utxos_json = this.utxos.map(i => i.getDict());
+        const data = {
+            "utxos": utxos_json,
+            "receiver_public_keys": this.receiver_public_keys,
+            "messages": this.messages
+        };
+        return data;
```
Comment (rkokhatskyi) [inline src/transaction.js:94]: you can return object directly like you did in other class

**Review by rkokhatskyi — APPROVED** [review]
LGTM 👍 


left few minor comments to consider in the next PR

**Pull request merged**